identity_credential/src/credential/status.rs

```diff
@@ -7,11 +7,63 @@ use serde::Serialize;
 use identity_core::common::Object;
 use identity_core::common::Url;
 
+#[cfg(feature = "revocation-bitmap")]
+use crate::revocation::status_list_2021::StatusList2021Entry;
+
+/// Credential status interface
+pub trait CredentialStatus {
+  /// `credentialStatus.type` field
+  fn r#type(&self) -> &str;
+  /// `credentialStatus.id` field
+  fn id(&self) -> &Url;
+}
+
 /// Information used to determine the current status of a [`Credential`][crate::credential::Credential].
 ///
 /// [More Info](https://www.w3.org/TR/vc-data-model/#status)
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum Status<T = Object> {
+  /// CredentialStatus using [`StatusList2021Entry`]
+  #[cfg(feature = "revocation-bitmap")]
+  StatusList2021(StatusList2021Entry),
+  /// Any other status
+  Other(CustomStatus<T>),
+}
+
+impl<T> From<CustomStatus<T>> for Status<T> {
+  fn from(value: CustomStatus<T>) -> Self {
+    Status::Other(value)
+  }
+}
+
+#[cfg(feature = "revocation-bitmap")]
+impl<T> From<StatusList2021Entry> for Status<T> {
+  fn from(value: StatusList2021Entry) -> Self {
+    Status::StatusList2021(value)
+  }
+}
+
+impl CredentialStatus for Status {
+  fn id(&self) -> &Url {
+    match self {
+      #[cfg(feature = "revocation-bitmap")]
+      Self::StatusList2021(s) => s.id(),
+      Self::Other(s) => s.id(),
+    }
+  }
+  fn r#type(&self) -> &str {
+    match self {
+      #[cfg(feature = "revocation-bitmap")]
+      Self::StatusList2021(s) => s.r#type(),
+      Self::Other(s) => s.r#type(),
+    }
+  }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
-pub struct Status<T = Object> {
```

Comment:
That's true, how should I address this?

Reply:
Maybe change the name of the current `Status` to `StatusType` or something similar and keep the custom one as `Status`. it doesn't look great but it's better than breaking IMO. Maybe @eike-hass can have an opinion.

Reply:
@abdulmth, the latest commit refactor things a bit (basically reverting all changes to `Status`). It's a bit annoying because the status of a parsed credential loses its type-safeness but at least this won't cause a breaking change